src/Libm.jl

```diff
@@ -1,5 +1,80 @@
 module Libm
 
-# package code goes here
+export xexp
 
-end # module
+const PI4_A = 0.78539816290140151978
+const PI4_B = 4.9604678871439933374e-10
+const PI4_C = 1.1258708853173288931e-18
+const PI4_D = 1.7607799325916000908e-27
+
+const M_4_PI = 1.273239544735162542821171882678754627704620361328125
+
+const L2U = .69314718055966295651160180568695068359375
+const L2L = .28235290563031577122588448175013436025525412068e-12
+const R_LN2 = 1.442695040888963407359924681001892137426645954152985934135449406931
+
+@inline double_to_raw_long_bits(d::Float64) = reinterpret(Int64, d)
+
+@inline long_bits_to_double(i::Int64) = reinterpret(Float64, i)
+
+@inline xfabs(x::Float64) = long_bits_to_double((0x7fffffffffffffff % Int64) & double_to_raw_long_bits(x))
+
+@inline function mulsign(x::Float64, y::Float64)
+  return long_bits_to_double(double_to_raw_long_bits(x)^(double_to_raw_long_bits(y) & (1 % Int64 )<< 63))
+end
+
+@inline sign(d::Float64) =  mulsign(1.0, d)
+@inline mla(x::Number, y::Number, z::Number) = x * y + z
+@inline xrint(x::Float64) = unsafe_trunc(Int32, round(x)) # in sleef, but this is a very limited way to truncate! (fixme)
+
+@inline xisnan(x::Float64) = (x != x) % Int32 
+@inline xisinf(x::Float64) = (x == Inf || x == -Inf) % Int32
+@inline xisminf(x::Float64) = (x == -Inf) % Int32
+@inline xispinf(x::Float64) = (x == Inf)  % Int32
+
+@inline function pow2i(q::Int32)
+  return long_bits_to_double(((q + 0x3ff) % Int64) << 52)
+end
+
+@inline function ldexpk(x::Float64, q::Int32)
+  m = q >> 31
+  m = (((m + q) >> 9) - m) << 7
+  q = q - (m << 2)
+  m += 0x3ff
+  m = m < 0 ? 0 : m
+  m = m > 0x7ff ? 0x7ff : m
+  u = long_bits_to_double((m % Int64) << 52)
+  x = x * u * u * u * u
+  u = long_bits_to_double(((q + 0x3ff) % Int64) << 52)
+  return x * u
+end
+
+
+function xexp(d::Float64)
+    q = xrint(d * R_LN2)
+
+    s = mla(q, -L2U, d)
+    s = mla(q, -L2L, s)
+
+    u = 2.08860621107283687536341e-09;
+    u = mla(u, s, 2.51112930892876518610661e-08)
+    u = mla(u, s, 2.75573911234900471893338e-07)
+    u = mla(u, s, 2.75572362911928827629423e-06)
+    u = mla(u, s, 2.4801587159235472998791e-05)
+    u = mla(u, s, 0.000198412698960509205564975)
+    u = mla(u, s, 0.00138888888889774492207962)
+    u = mla(u, s, 0.00833333333331652721664984)
+    u = mla(u, s, 0.0416666666666665047591422)
+    u = mla(u, s, 0.166666666666666851703837)
+    u = mla(u, s, 0.5)
```

Comment:
Thanks for your careful eyes. I'll do that as soon as I have more function ported.